cilium/host_map.cc

```diff
@@ -149,14 +148,14 @@ PolicyHostMap::PolicyHostMap(ThreadLocal::SlotAllocator& tls)
 // This is used in production
 PolicyHostMap::PolicyHostMap(Server::Configuration::CommonFactoryContext& context)
     : PolicyHostMap(context.threadLocal()) {
-  scope_ = context.scope().createScope(name_);
+  scope_ = context.serverScope().createScope(name_);
 }
 
 void PolicyHostMap::startSubscription(Server::Configuration::CommonFactoryContext& context) {
-  subscription_ =
-      subscribe("type.googleapis.com/cilium.NetworkPolicyHosts", context.localInfo(),
-                context.clusterManager(), context.mainThreadDispatcher(),
-                context.api().randomGenerator(), *scope_, *this, this->shared_from_this());
+  subscription_ = subscribe("type.googleapis.com/cilium.NetworkPolicyHosts", context.localInfo(),
+                            context.clusterManager(), context.mainThreadDispatcher(),
+                            context.api().randomGenerator(), *scope_, *this,
+                            std::make_shared<Cilium::PolicyHostDecoder>());
```

Comment:
resource decoder is passed to `GrpcSubscriptionImpl()` which takes a shared pointer, so this needs to be a shared pointer. These maps are singletons, so that at any time there is only one of them, so there is no opportunity for sharing either.